lib/src/types.ts

```diff
@@ -134,6 +141,12 @@ export interface UnionType {
   discriminator?: string;
 }
 
+export interface IntersectionType {
+  kind: TypeKind.INTERSECTION;
+  types: Type[];
+  discriminator?: string;
```

Comment:
Perhaps we can remove `discriminator` for `IntersectionType`. There doesn't appear to be any good use case for it.